### PR TITLE
feat: added firebase authentication in social media app

### PR DESCRIPTION
**Pull request by kedark152**
### feat: added firebase authentication in social media app

Features added:-
- Used Redux Tool Kit for managing Authentication States & user data.
- Create New User (Signup) using firebase authentication and update user data in firestore database.
- Added Firebase Authentication for Login, Logout.
- Added firebase onAuthStateChange feature for Auth Persistance after page refresh.
- Added RequiresAuth for page redirection based on Current Auth State.
- used chakra ui toast features to show toast on success/error in authentication.
- Added Popover box for logout confirmation in Sidebar Menu.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -21,7 +22,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React App</title>
+    <title>AgroLife - Social Media App for Farming Community</title>
```
Comment (pariyar07) [inline public/index.html:25]: Great niche idea.

File: src/app/store.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { configureStore } from '@reduxjs/toolkit';
+import authReducer from '../features/auth/authSlice';
```
Comment (pariyar07) [inline src/app/store.js:2]: Use absolute imports. Makes it easy to manage imports.

**Review by pariyar07 — COMMENTED** [review]
Everything looks great. Few tweaks and you are good to merge. There is some problem with your post image on the homepage its not getting displayed.

**Review by simran61 — APPROVED** [review]

**Review by t007rushi — APPROVED** [review]

**Comment by kedark152** [comment]
Thanks all for Review will make suggested changes next time.

**Pull request merged**